### DISCUSSION TITLE
Best Practice for Handling FormData with Files and Booleans in ZSA

### DISCUSSION THREAD

**Discussion by yossielimelech** [IdoPesok/zsa#186] [Q&A]
I'm using ZSA and must submit both files and boolean values (Switch control / shadcn) in a single server action. ZSA restricts the input type to either formData or a normal object. Here's a simplified example of my use case:

```
export const updateNotification = authedProcedure.createServerAction()
    .input(
        updateNotificationSchema.extend({
            file: z.instanceof(File).optional(),
            active: z.boolean(),
        }),
        {
            type: "formData", // Required for file, converts 'active' to string
        }
    )
    .handler(async ({ input }) => {
        // handler logic
    });

```
    
How can I manage both data types in one request without resorting to two separate calls?
I'm probably missing something very basic here.

Thanks!

**Comment by IdoPesok** [comment]
Hi, I think [binding](https://zsa.vercel.app/docs/forms#binding-arguments) is what you are looking for.

**Reply by yossielimelech** [reply]
Tyvm! I don't know how I missed it!
